.github/workflows/test.yml

```diff
@@ -86,12 +86,6 @@ jobs:
         if: matrix.os == 'windows-latest'
         run: continuous_integration\test_script_windows.bat
 
-      - name: Publish Test Report
```

Comment:
I'm not sure what it does, really, but why did you remove this action in this PR?

Reply:
I removed it because it's failing in some PR and nobody fixes it.
Moreover, the version of git action is quite old; version 5 should be used.

This is a reason for some warnings in the summary of the GitHub action:
![image](https://github.com/user-attachments/assets/36f665cc-b890-4a0d-b910-18dd98a4a59e)